tests/unit/jobserver/test_auth_pipeline.py

```diff
@@ -1,22 +1,45 @@
-from jobserver.auth_pipeline import (
-    notify_on_new_user,
-    set_fullname,
-    set_notifications_email,
-)
-
-from ...factories import UserFactory
-
-
-def test_notify_on_new_user_with_existing_user(slack_messages):
-    user = UserFactory()
-    notify_on_new_user(user, is_new=False)
-    assert len(slack_messages) == 0
-
-
-def test_notify_on_new_user_with_new_user(slack_messages):
-    user = UserFactory()
+from jobserver.auth_pipeline import pipeline
+
+from ...factories import UserFactory, UserSocialAuthFactory
+
+
+def test_pipeline():
```

Comment:
Might be more expressive to call this something like:

`test_pipeline_with_existing_user()`

Reply:
It also tests with a new user too! This was originally two tests, one with that name, and I realised the existing user test was duplicating the entirety of the first test.

Reply:
Are you thinking of `test_pipeline_new_user`, which seems to do what you're describing? I'm not sure I see the advantage of having both tests check the behaviour for an existing user. My preference would be to have it in a separate test rather than have one test check both things.

Reply:
I think I meant to combine them into one test and must have gotten distracted half way 😂  I'll split them up 👍